StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/Embedded/EmbeddedPaymentElement.swift

```diff
@@ -133,54 +133,26 @@ public final class EmbeddedPaymentElement {
             }
 
-            // Store the old payment option before we update self.formViewController
-            let oldPaymentOption = self?.paymentOption
-
-            // 2.1. Re-initialize embedded form view controller to update the UI to match the newly loaded data.
-            if let formPaymentMethodType = self?.selectedFormViewController?.selectedPaymentOption?.paymentMethodType {
-                self?.selectedFormViewController = EmbeddedFormViewController(configuration: configuration,
-                                                                intent: loadResult.intent,
-                                                                elementsSession: loadResult.elementsSession,
-                                                                shouldUseNewCardNewCardHeader: loadResult.savedPaymentMethods.first?.type == .card,
-                                                                paymentMethodType: formPaymentMethodType,
-                                                                previousPaymentOption: self?.selectedFormViewController?.selectedPaymentOption,
-                                                                analyticsHelper: analyticsHelper)
-
-            }
-
-            // 2.4 Re-initialize embedded view to update the UI to match the newly loaded data.
-            let embeddedPaymentMethodsView = Self.makeView(
+            // 2. At this point, we're still the latest update and update is successful - update self properties and inform our delegate.
+            let previousPaymentOption = self._paymentOption
+            let previousDisplayData = self.paymentOption
+            self.loadResult = loadResult
+            self.savedPaymentMethods = loadResult.savedPaymentMethods
+            self.formCache = .init() // Clear the cache because the form may have changed e.g. different mandate or different fields.
+            self.embeddedPaymentMethodsView = Self.makeView(
                 configuration: configuration,
                 loadResult: loadResult,
                 analyticsHelper: analyticsHelper,
-                previousPaymentOption: self?._paymentOption,
+                previousPaymentOption: previousPaymentOption,
                 delegate: self
             )
-
-            // 3. Pre-load image into cache
-            // Call this on a detached Task b/c this synchronously (!) loads the image from network and we don't want to block the main actor
-            let fetchPaymentOption = Task.detached(priority: .userInitiated) {
-                // This has the nasty side effect of synchronously downloading the image (see https://jira.corp.stripe.com/browse/MOBILESDK-2604)
-                // This caches it so that DownloadManager doesn't block the main thread when the merchant tries to access the image
-                return await embeddedPaymentMethodsView.selection?.paymentMethodType?.makeImage(updateHandler: nil)
-            }
-            _ = await fetchPaymentOption.value
```

Comment:
I realized we don't need to do this; there's still a bad problem where `paymentOption` will block the main thread if the image isn't loaded, but it can't happen in practice b/c either:
1. We're preserving the old paymentOption, in which case the image was already loaded
2. The payment option changed to the default, which is always of a type we have a hardcoded image for (saved PM, apple pay, link).